integrations/slu-lex/src/LexSlu.ts

```diff
@@ -0,0 +1,170 @@
+    });
+  }
+
+  getDefaultConfig(): LexSluConfig {
+    return {
+      ...super.getDefaultConfig(),
+      bot: { id: '', aliasId: '' },
+      region: '',
+      credentials: {
+        accessKeyId: '',
+        secretAccessKey: '',
+      },
+      fallbackLocale: 'en_US',
+      asr: true,
+      nlu: true,
+    };
+  }
+
+  getInitConfig(): RequiredOnlyWhere<LexSluConfig, 'credentials' | 'region' | 'bot'> {
+    return {
+      bot: { id: '', aliasId: '' },
+      region: '',
+      credentials: {
+        accessKeyId: '',
+        secretAccessKey: '',
+      },
+    };
+  }
+
+  async processAudio(jovo: Jovo, audio: ParsedAudioInput): Promise<AsrData | undefined> {
+    if (!this.config.asr || !jovo.$session.id) {
+      return;
+    }
+    const params: RecognizeUtteranceCommandInput = {
+      botId: this.config.bot.id,
+      botAliasId: this.config.bot.aliasId,
+      requestContentType: `audio/x-l16; sample-rate=${audio.sampleRate}; channel-count=1`,
+      inputStream: audio.toWav(),
+      localeId: this.getLocale(jovo),
+      sessionId: jovo.$session.id,
+    };
+
+    const command = new RecognizeUtteranceCommand(params);
+    const response = await this.client.send(command);
+    if (!response.inputTranscript) {
+      return;
+    }
+    // The return inputTranscript ist a gzipped string that is encoded with base64
+    // base64 -> gzip
+    const transcriptBuffer = Buffer.from(response.inputTranscript, 'base64');
+    // gzip -> string
+    const textBuffer = await asyncGunzip(transcriptBuffer);
+    // The string of textBuffer will always contain double quotes, therefore we can parse it with JSON to get rid of it.
+    const parsedText = JSON.parse(textBuffer.toString());
+    return {
+      text: parsedText,
+    };
+  }
+
+  async processText(jovo: Jovo, text: string): Promise<NluData | undefined> {
+    if (!this.config.nlu || !jovo.$session.id) {
+      return;
+    }
+    const params: RecognizeTextCommandInput = {
+      botId: this.config.bot.id,
+      botAliasId: this.config.bot.aliasId,
+      text,
+      localeId: this.getLocale(jovo),
+      sessionId: jovo.$session.id,
+    };
+    const command = new RecognizeTextCommand(params);
+    const response = await this.client.send(command);
+    if (!response.interpretations) {
+      return;
+    }
+    // Assuming the interpretations will be sorted by confidence,
+    return this.getNluDataFromInterpretation(response.interpretations[0]);
+  }
+
+  private getNluDataFromInterpretation(interpretation: Interpretation): NluData | undefined {
+    if (!interpretation.intent) {
+      return;
+    }
+
+    const nluData: NluData = {
+      intent: interpretation.intent.name,
+    };
+    if (interpretation.intent.slots) {
+      nluData.entities = Object.entries(interpretation.intent.slots).reduce(
+        (entities: EntityMap, [name, slot]) => {
+          if (!slot.value) {
+            return entities;
+          }
+          entities[name] = {
```

Comment:
Is it possible that `id` is missing here? Should be the same as `resolved` if Lex doesn't offer an `id`

Reply:
Thanks for pointing that out. The `id` is now properly set as well.
Lex sadly doesn't have a specific id, therefore it will be the same value as `resolved`.